src/sagas/repository.ts

```diff
@@ -1023,6 +1028,40 @@ export function* rejectSnapshotAccessRequest({ payload }: any): any {
   }
 }
 
+export function* approveSnapshotAccessRequest({ payload }: any): any {
+  const {
+    snapshotAccessRequestResponse: { id, snapshotName },
+  } = payload;
+  const snapshotRequest: SnapshotRequestModel = {
+    name: generateSnapshotNameFromAccessRequestInformation(id, snapshotName),
+    description: `Snapshot created from ${snapshotName}`,
+    contents: [
+      {
+        datasetName: 'unused',
+        mode: SnapshotRequestContentsModelModeEnum.ByRequestId,
+        requestIdSpec: {
+          snapshotRequestId: id,
+        },
+      },
+    ],
+  };
+  try {
+    yield call(authPut, `/api/repository/v1/snapshotAccessRequests/${id}/approve`);
```

Comment:
I see approve called here, and the payload for the create endpoint constructed above, but where is the create endpoint actually called?

Reply:
The line below! https://github.com/DataBiosphere/jade-data-repo-ui/pull/1664/files#diff-b0cc26de535914428b254ecf0a5ed0e23a1c0e9753bbcb59a2af7167f6cd37ceR1050